vendor/bat-native-ledger/src/bat/ledger/internal/media/vimeo.cc

```diff
@@ -83,13 +85,18 @@ std::string Vimeo::GetIdFromVideoPage(const std::string& data) {
 }
 
 // static
-std::string Vimeo::GenerateFaviconUrl(const std::string& id) {
-  if (id.empty()) {
-    return "";
-  }
+std::string Vimeo::GenerateFaviconUrl(const std::string& data) {
+    if (data.empty()) {
+        return "";
+    }
 
-  return base::StringPrintf("https://i.vimeocdn.com/portrait/%s_300x300.webp",
-                            id.c_str());
+    std::string portraitUrlID = braveledger_media::ExtractData(data,
+            "src_2x\":\"https:\\/\\/i.vimeocdn.com\\/portrait\\/", "_");
+
+    std::string favIconUrl = base::StringPrintf("https://i.vimeocdn.com/portrait/%s_300x300.webp",
```

Comment:
we need to first check if `portraitUrlID` is maybe empty and return empty in this case so that we don't return corrupted url

Reply:
Good catch - thanks!

Reply:
Fixed